drivers/device_uid/stm32_uid.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2018 Alexander Wachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <soc.h>
+#include <device_uid.h>
+#include <string.h>
+
+#if !defined(CONFIG_SOC_SERIES_STM32F0X) && \
+	!defined(CONFIG_SOC_SERIES_STM32F3X) && \
+	!defined(CONFIG_SOC_SERIES_STM32L4X) && \
+	!defined(CONFIG_SOC_SERIES_STM32F7X)
+#error UNIQUE ID only available on STM32F0, STM32F3, STM32l4 and STM32F7 series
+#endif
+
+struct stm32_uid {
+	u32_t id[3];
+};
+
+static int stm32_uid_init(struct device *dev)
+{
+	return 0;
+}
+
+static size_t stm32_uid_get(struct device *dev, u8_t *buffer, size_t length)
+{
+	struct stm32_uid dev_id;
+
+	if (buffer == NULL && length == 0) {
+		return sizeof(dev_id.id);
```

Comment:
documentation says that function should return
"size of the device UID copied"

was the intention to return the expected length much like snprintf does? (i.e. if buffer it too short inform how much is really needed)?

Reply:
ok - I read the comments in the API part.
If we want to have posix way of length return then documentation must be updated and this code fixed for case when `(lenght < sizeof(dev_id.id))`

Reply:
I think I will remove the get length function.